invenio_vocabularies/services/config.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+# Copyright (C) 2024 Uni Münster.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Vocabulary services configs."""
+
+import sqlalchemy as sa
+from flask import current_app
+from invenio_i18n import lazy_gettext as _
+from invenio_records_resources.services import (
+    Link,
+    LinksTemplate,
+    RecordService,
+    RecordServiceConfig,
+    SearchOptions,
+    pagination_links,
+)
+from invenio_records_resources.services.base import (
+    ConditionalLink,
+    Service,
+    ServiceListResult,
+)
+from invenio_records_resources.services.records.components import DataComponent
+from invenio_records_resources.services.records.params import (
+    FilterParam,
+    SuggestQueryParser,
+)
+
+from ..records.api import Vocabulary
+from ..records.models import VocabularyType
+from . import results
+from .components import PIDComponent, VocabularyTypeComponent
+from .permissions import PermissionPolicy
+from .schema import TaskSchema, VocabularySchema
+
+
+def is_custom_vocabulary_type(vocabulary_type, context):
+    """Check if the vocabulary type is a custom vocabulary type."""
+    return vocabulary_type["id"] in current_app.config.get(
+        "VOCABULARIES_CUSTOM_VOCABULARY_TYPES", []
+    )
+
+
+class VocabularySearchOptions(SearchOptions):
+    """Search options for vocabularies."""
+
+    params_interpreters_cls = [
+        FilterParam.factory(param="tags", field="tags"),
+    ] + SearchOptions.params_interpreters_cls
+
+    suggest_parser_cls = SuggestQueryParser.factory(
+        fields=[
```

Comment:
Question:
Can you explain why these specific fields and weights boosting were chosen for the suggest_parser_cls? 
Documenting the rationale behind this decision for future reference would be helpful as a small comment in the class docstring.

Reply:
I did not add this code, so someone else would have to answer the question. I simply split the service and the Config into two separate files (this is actually done in #310, that's why I recommend merging that PR first and then rebasing this branch).

Reply:
I'll resolve this conversation for this PR, since it is not related to changes made here. If there is a problem with indexing you could create a new issue.